From 4dbb2748f3f7d19941794f2e1993b64d1ece542c Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Mon, 2 Jul 2018 22:01:43 +0200
Subject: [PATCH 1/2] [device] ::Huawei Kirin 960:: accept broken rpfilter
 match

How bad a security flaw is this?
People lived with rpfilter on IPv4 for a very long time...

Change-Id: I9aa63d18e54a8254133adf97bf757c03d6b66757
---
 server/TetherController.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/TetherController.cpp b/server/TetherController.cpp
index 0c5b6bfe..d9439be6 100644
--- a/server/TetherController.cpp
+++ b/server/TetherController.cpp
@@ -696,7 +696,7 @@ int TetherController::setForwardRules(bool add, const char *intIface, const char
         "*raw\n"
         "%s %s -i %s -m rpfilter --invert ! -s fe80::/64 -j DROP\n"
         "COMMIT\n", op, LOCAL_RAW_PREROUTING, intIface);
-    if (iptablesRestoreFunction(V6, rpfilterCmd, nullptr) == -1 && add) {
+    if (iptablesRestoreFunction(V6, rpfilterCmd, nullptr) == -1 && add && false) {
         return -EREMOTEIO;
     }
 
-- 
2.17.1

